scripts/cleanup.sh

```diff
@@ -0,0 +1,29 @@
+## Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.## 
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0 
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## 
+
+#! /usr/bin/env bash
+set +x
+
+export DEBIAN_FRONTEND=noninteractive
+
+if [ ! -d /github/workspace/.cache ]; then
+    echo "Cache directory not found in /workspace"
+    exit 1
```

Comment:
Under which condiditions could this occur? Should the cleanup fail in this case?

Reply:
Looking at the code below, the key and kubeconfig are created before the cache. Does this mean they should be deleted first?

Reply:
the thing is, if cache is not found, we don't have info to delete the cloud instance. Then the deletion order does not matter much. we need to delete all during clean up stage.

Reply:
Should we not ALWAYS try to remove the `key.pem` and `kubeconfig`. 

Also, why are we exiting with a non-zero code?

Reply:
if there's no cache file we need to report and error (exit 1) because without it, we can perform the clean up.
we should always remove all 3 files. key is the most important one, the other 2 are useless once the ec2 instance is down.

Reply:
@elezar PTAL